src/repl-top-level-await.ts

```diff
@@ -0,0 +1,309 @@
+// Based on https://github.com/nodejs/node/blob/975bbbc443c47df777d460fadaada3c6d265b321/lib/internal/repl/await.js
```

Comment:
When we incorporate various bits of node's source code, we traditionally keep it in dist-raw.  The goal is to make as few modifications as possible so that merging upstream changes is easier.  The goal is to reduce the maintenance burden in the future.  Should we do the same here?

Reply:
Yes sure, probably just need to add missing `node-primordials` and adjust `acorn` imports, besides from that it should be equal.

Reply:
Ok, that'd be great.  I'm not very familiar with acorn nor with the version that is bundled within node.  I see https://www.npmjs.com/package/acorn-node, which claims to be a version of acorn pre-configured to support all the same language features as node.  Is that something we need to do here?

I admitted haven't gone through your code in detail, so this may be obvious from the code.

Reply:
That package size is greater (`3.4M    node_modules`) and also doesn't seem to match acorn versions used https://github.com/nodejs/node/tree/master/deps/acorn (last version in node was published 13 days ago, while `acorn-node` last version is 1y old). The other thing we can do is copy `acorn.js` & `walk.js` from node `deps/acorn` and paste them under `dist-raw/deps`, size would be also less (208K + 16K)

Reply:
I've tested the above and confirm it works correctly, I'm sure that's the path of least maintenance burden while also remaining fully compatible. Chages:

- Copy `primordials` as `dist-raw/node-primordials` (cherry picking primordials is a lot more work and may not be accurate, the ~10Kb save is not worth it)
- Copy `deps/acorn/*` as `dist-raw/deps/acorn/*`
- Copy `await.js` as `dist-raw/node-repl-await`
- Adjust imports in new `dist-raw` files
- Create `globalThis.primordials = {}` at the start of `primordials` file

Only piece left is where to preload `primordials` file.

Reply:
I've pushed a commit that does the above.